File: 03-poo/13-excepciones-personalizadas/scala/ejemplos/ExcepcionDominio.scala
class SaldoInsuficienteException(mensaje: String) extends Exception(mensaje)
class CuentaSaldo(private var saldo: Double):
  def retirar(monto: Double): Unit =
    if monto > saldo then throw new SaldoInsuficienteException("Saldo insuficiente")
    saldo -= monto

@main def excepcionDominio(): Unit =
  try new CuentaSaldo(50).retirar(80)
  catch case e: SaldoInsuficienteException => println(e.getMessage)

File: 03-poo/13-excepciones-personalizadas/java/ejemplos/ExcepcionDominio.java
public class ExcepcionDominio {
    static class SaldoInsuficienteException extends Exception {
        SaldoInsuficienteException(String mensaje){ super(mensaje); }
    }
    static class Cuenta {
        double saldo=50;
        void retirar(double monto) throws SaldoInsuficienteException {
            if (monto > saldo) throw new SaldoInsuficienteException("Saldo insuficiente");
            saldo -= monto;
        }
    }
    public static void main(String[] args){ try { new Cuenta().retirar(80); } catch (SaldoInsuficienteException e){ System.out.println(e.getMessage()); } }
}
